packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -6,6 +6,74 @@ import 'package:flutter/widgets.dart';
 import 'bottom_tab_bar.dart';
 import 'theme.dart';
 
+/// Coordinates tab selection between a [CupertinoTabBar] and a [CupertinoTabScaffold].
+///
+/// The [currentIndex] property is the index of the selected tab, changing its value updates
+/// the actively displayed tab of the [CupertinoTabScaffold] the [CupertinoTabController] controls,
+/// as well as the currently selected tab item of its [CupertinoTabBar].
+///
+/// There're 2 ways to programmatically change the actively displayed tab of a [CupertinoTabScaffold]
+/// (and the selected tab item of the [CupertinoTabBar]):
+/// 1. Change the [currentIndex] of the existing controller of the [CupertinoTabScaffold],
+///    see the example below.
+/// 2. Rebuild the [CupertinoTabScaffold] widget with a new [CupertinoTabController], whose
+///    [currentIndex] is set to the new value.
+///
+/// {@tool sample}
+///
+/// `CupertionoTabController` can be used to switch tabs from ancestor widgets:
+///
+/// ```dart
+/// class MyCupertinoTabScaffoldPage extends StatefulWidget {
+///   @override
+///   _CupertinoTabScaffoldPageState createState() => _CupertinoTabScaffoldPageState();
+/// }
+///
+/// class _CupertinoTabScaffoldPageState extends State<MyCupertinoTabScaffoldPage> {
+///   final CupertinoTabController _controller = CupertinoTabController();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return CupertinoTabScaffold(
+///       tabBar: CupertinoTabBar(
+///         items: <BottomNavigationBarItem> [
+///           // ...
+///         ]
+///       ),
+///       controller: _controller,
+///       tabBuilder: (BuildContext context, int index) {
+///         return Center(
+///           child: CupertinoButton(
+///             child: const Text('Go to first tab'),
+///             onPressed: () => _controller.currentIndex = 0
+///           )
+///         );
+///       }
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// see also:
+/// [CupertinoTabScaffold], a tabbed application root layout that can be controlled by a [CupertinoTabController].
+class CupertinoTabController extends ValueNotifier<int> {
+  /// Creates an object that manages the index of the current selected tab required by a [CupertinoTabScaffold].
```

Comment:
It's not 'required by a [CupertinoTabScaffold]'. 

Just `Creates a [CupertinoTabController] to control the tab index of [CupertinoTabScaffold] and [CupertinoTabBar].`